packages/ag-charts-community/src/dom/elements.ts

```diff
@@ -8,13 +8,23 @@ interface MissingAriaAttrs {
     ariaControls?: string;
 }
 
-export interface ButtonOptions {
-    label: string | HTMLElement;
+// These types force a compilation error if the developer tries to add an icon-only
+// menu item without an accessible text alternative.
+type LabelAndIcon = { label: string; icon?: AgIconName; altText?: undefined };
+type IconOnly = { label?: undefined; icon: AgIconName; altText: string };
+export type LabelIcon = LabelAndIcon | IconOnly;
```

Comment:
Note: This `LabelIcon` allows 3 syntax modes:

```ts
{ label }         // text only
{ label, icon }   // text & icon
{ icon, altText } // icon only (with altText for screenreaders)
```

Specifying an `icon` without `altText`, or specifying both a `label` and `altText` will raise an compiler error.